hedera-node/src/main/java/com/hedera/services/context/ServicesContext.java

```diff
@@ -570,7 +569,9 @@
 	private AtomicReference<FCMap<MerkleEntityId, MerkleSchedule>> queryableSchedules;
 	private AtomicReference<FCMap<MerkleBlobMeta, MerkleOptionalBlob>> queryableStorage;
 	private AtomicReference<FCMap<MerkleEntityAssociation, MerkleTokenRelStatus>> queryableTokenAssociations;
-	private AtomicReference<FCInvertibleHashMap<MerkleUniqueTokenId, MerkleUniqueToken, OwnerIdentifier>> queryableUniqueTokens;
+	private AtomicReference<FCMap<MerkleUniqueTokenId, MerkleUniqueToken>> queryableUniqueTokens;
+	private FCOneToManyRelation<EntityId, MerkleUniqueTokenId> uniqueTokenOwnership = new FCOneToManyRelation<>();
+	private FCOneToManyRelation<EntityId, MerkleUniqueTokenId> uniqueTokenAssociations = new FCOneToManyRelation<>();
```

Comment:
We should place the new structures inside the `TypedTokenStore` and `rebuildViews` on `rebuildStoreViewsIfPresent`